operator/cmd/root.go

```diff
@@ -646,13 +647,26 @@ func (legacy *legacyOnLeader) onStart(_ hive.HookContext) error {
 	}
 
 	if legacy.clientset.IsEnabled() {
-		if operatorOption.Config.EndpointGCInterval != 0 {
+		if operatorOption.Config.EndpointGCInterval != 0 && !option.Config.DisableCiliumEndpointCRD {
 			enableCiliumEndpointSyncGC(legacy.ctx, &legacy.wg, legacy.clientset, ciliumNodeSynchronizer, false)
 		} else {
-			// Even if the EndpointGC is disabled we still want it to run at least
+			// Even if the EndpointGC or CEPs are disabled, we still want it to run at least
 			// once. This is to prevent leftover CEPs from populating ipcache with
 			// stale entries.
-			enableCiliumEndpointSyncGC(legacy.ctx, &legacy.wg, legacy.clientset, ciliumNodeSynchronizer, true)
+			// First check if CEP CRD is available, as it could be missing if CEPs are not enabled.
```

Comment:
The context in the commit msg was very helpful to understand this code so could you elaborate here as well? I was initially confused until I went back and read your commit msg.

Reply:
Ah for sure, thank you for bringing that up. I'll change my wording around to make it more clear and add the info from my commit message as a comment. From my understanding as well, this function doesn't do any GC for kvstore mode- this is taken care of by leases in etcd if I'm not mistaken.

Reply:
Ah right--leases. Thanks!